api/tests/unit/segments/test_unit_segments_models.py

```diff
@@ -167,3 +169,247 @@ def test_condition_get_update_log_message(segment, segment_rule, mocker):
 )
 def test_segment_id_exists_in_rules_data(rules_data, expected_result):
     assert Segment.id_exists_in_rules_data(rules_data) == expected_result
+
+
+def test_deep_clone_of_segment(
+    project: Project,
+    feature: Feature,
+) -> None:
+    # Given
+    segment = Segment.objects.create(
+        name="SpecialSegment",
+        description="A lovely, special segment.",
+        project=project,
+        feature=feature,
+    )
+
+    # Check that the versioning is correct, since we'll testing
+    # against it later in the test.
+    assert segment.version == 1
+    assert segment.version_of == segment
+
+    parent_rule1 = SegmentRule.objects.create(
+        segment=segment, type=SegmentRule.ALL_RULE
+    )
+    parent_rule2 = SegmentRule.objects.create(
+        segment=segment, type=SegmentRule.ANY_RULE
+    )
+
+    child_rule1 = SegmentRule.objects.create(
+        rule=parent_rule1, type=SegmentRule.ANY_RULE
+    )
+    child_rule2 = SegmentRule.objects.create(
+        rule=parent_rule1, type=SegmentRule.ALL_RULE
+    )
+    child_rule3 = SegmentRule.objects.create(
+        rule=parent_rule2, type=SegmentRule.NONE_RULE
+    )
+    child_rule4 = SegmentRule.objects.create(
+        rule=parent_rule2, type=SegmentRule.ALL_RULE
+    )
```

Comment:
I think we can stick to the structure created by the dashboard here which is just: 

```
parent_rule = SegmentRule.objects.create(
    segment=segment, type=SegmentRule.ALL_RULE,
)

child_rule1 = SegmentRule.objects.create(
    rule=parent_rule, type=SegmentRule.ANY_RULE
)
child_rule2 = SegmentRule.objects.create(
    rule=parent_rule, type=SegmentRule.NONE_RULE
)
...
```

Note that the UI only allows 'any' and 'none' rules nested beneath a single 'all' rule.

Reply:
Sounds good. Fixed locally.